src/coreclr/vm/fcall.h

```diff
@@ -616,11 +623,21 @@ LPVOID __FCThrowArgument(LPVOID me, enum RuntimeExceptionKind reKind, LPCWSTR ar
 
 #define HELPER_METHOD_FRAME_END_EX(gcpoll,allowGC)                          \
             UNINSTALL_UNWIND_AND_CONTINUE_HANDLER;                          \
+            };                                                              \
+            auto nonCapturingLambda = [](uintptr_t pointerToLambda) { \
+                (*(decltype(helperFrameLambda)*)pointerToLambda)(); \
+            }; \
+            CallOnOSThread(nonCapturingLambda, (uintptr_t)&helperFrameLambda); \
```

Comment:
I assume I haven't gotten far enough past this PR... while obviously you need a function pointer to pass off, why do you need the two layers (or "functionpointer + pointer arg" as in the PR title")?  Is this setting up for a future where the wrapper function can be replaced by something else?

Reply:
Oh, that's a peculiarity of C++ lambdas. You cannot get a normal function pointer to a lambda which captures any variables. However, I need that for the TransitionToOSThread function. So we have a 2 level structure where the functionpointer passed to the CallOSThread function is a captureless lambda, but we can then call through to the inner lambda with all the args in place. (Its legal to have a pointer to any lambda that can be converted to/from void*). We need the captureless lambda, as that is needed to know the actual type of the "useful" lambda.